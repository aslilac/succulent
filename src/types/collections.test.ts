/// <reference types="jest" />
import { check, is, $object, $boolean, $Map, $number, $Set } from "../index";

test("$Map", () => {
	const example = new Map([
		[true, {}],
		[false, {}],
	]);

	expect(is(example, $Map($boolean, $object))).toBe(true);
	expect(is(example, $Map($number, $object))).toBe(false);
	expect(is(example, $Map($boolean, $number))).toBe(false);

	expect(() => check(example, $Map($boolean, $number))).toThrowErrorMatchingSnapshot();
});

test("$Set", () => {
	const example = new Set([true, false]);

	expect(is(example, $Set($boolean))).toBe(true);
	expect(is(example, $Set($number))).toBe(false);

	expect(() => check(example, $Set($number))).toThrowErrorMatchingSnapshot();
});
