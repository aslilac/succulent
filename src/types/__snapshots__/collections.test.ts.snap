// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`$Map 1`] = `
"Expected Map<boolean, number>, got [object Map]
  Expected number, got [object Object]"
`;

exports[`$Set 1`] = `
"Expected Set<number>, got [object Set]
  Expected number, got true"
`;
